source/nordic-sdk/components/libraries/bootloader_dfu/dfu_app_handler.c

```diff
@@ -67,52 +66,98 @@ static void dfu_app_reset_prepare(void)
 static void interrupts_disable(void)
 {
     uint32_t interrupt_setting_mask;
-    uint32_t irq = 0; // We start from first interrupt, i.e. interrupt 0.
+    uint32_t irq;
 
     // Fetch the current interrupt settings.
     interrupt_setting_mask = NVIC->ISER[0];
 
-    for (; irq < MAX_NUMBER_INTERRUPTS; irq++)
+    // Loop from interrupt 0 for disabling of all interrupts.
+    for (irq = 0; irq < MAX_NUMBER_INTERRUPTS; irq++)
     {
         if (interrupt_setting_mask & (IRQ_ENABLED << irq))
         {
-            // The interrupt was enabled, and hence disable it.
+            // The interrupt was enabled, hence disable it.
             NVIC_DisableIRQ((IRQn_Type)irq);
         }
     }
 }
 
-#if NEEDED
```

Comment:
I don't recall why I added the #if NEEDED here. Perhaps the original reason has been lost as we upgraded to newer SDKs.